### PR TITLE
ISPN-16759 Do not enable memcached text protocol implicitly without s…

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-16759

### CODE REVIEW

File: server/tests/src/test/java/org/infinispan/server/functional/memcached/MemcachedOperations.java [hunk 1]
```diff
@@ -105,13 +105,31 @@ public void testConcurrentGets(ConnectionFactoryBuilder.Protocol protocol) throw
...
+   @ParameterizedTest
+   @EnumSource(ConnectionFactoryBuilder.Protocol.class)
+   public void testImplicitNoAuthMemcached(ConnectionFactoryBuilder.Protocol protocol)  {
+      ConnectionFactoryBuilder builder = new ConnectionFactoryBuilder();
+      builder.setProtocol(protocol);
+      MemcachedClient client = SERVERS.memcached().withClientConfiguration(builder).withPort(11222).get();
+      try {
+         client.get(k());
+         if (protocol == ConnectionFactoryBuilder.Protocol.TEXT) {
+            throw new RuntimeException("Implicit memcached text request should have failed");
```
Comment (wburns) [inline server/tests/src/test/java/org/infinispan/server/functional/memcached/MemcachedOperations.java:117]: This exception can never be propagated due to the catch block. Should it not propagate? If not then I would say there is no reason to throw the exception.

Comment (tristantarrant) [reply server/tests/src/test/java/org/infinispan/server/functional/memcached/MemcachedOperations.java:118]: I've made the caught exception more specific

**Comment by tristantarrant** [comment]
@wburns can you recheck ?

**Review by wburns — APPROVED** [review]
LGTM, just want to rerun CI though as it was acting weird.

**Comment by jabolina** [comment]
Creating an issue for it.

**Review by jabolina — APPROVED** [review]

**Pull request merged**